### PR TITLE
podman support added

### PR DESCRIPTION
**Pull request by jwszolek**
## What
Podman was added as an alternative for running airbyte locally due to its rootless container capabilities. With Podman, users can execute containers without having to operate them as the root user, providing enhanced security to the system through reduced risk of privilege escalation attacks and other security vulnerabilities.

## How
In addition to the integration of Podman, a dedicated docker-compose file and an environmental file with specific configuration values have also been introduced. This provides greater flexibility and customization options for running Airbyte locally.

## Recommended reading order
1. docker-compose.podman.yaml
2. .env.podman
3. README.md

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

NO

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

### CODE REVIEW

**Comment by jwszolek** [comment]
hey @marcosmarxm,  sure! the Contributor License Agreement has been signed.

**Comment by terencecho** [comment]
I'm also not convinced it's best to add an almost identical env/docker-compose file.

Seems like something that should just be in our documentation that says, you can also use podman to deploy, all you have to do is change these handful of lines. Definitely something cool that we should share though and I can't imagine any objections if we're only changing the documentation.

I'm not sure if the platform enablement team is the best judge of this though. Not sure if this is more related to infra or platform-compose team.

**Comment by jwszolek** [comment]
Thank you for your feedback! On the surface, docker-compose appears similar, but requires modifications to have a fully functional podman-compose file. From my perspective, utilizing a pre-configured podman-compose file is more efficient, as it eliminates the need to spend time studying the documentation to understand how it differs from the traditional docker-compose file.

File: README.md [hunk 1]
```diff
@@ -44,14 +44,20 @@ Explore our [demo app](https://demo.airbyte.io/).
...
 ```bash
 git clone https://github.com/airbytehq/airbyte.git
 cd airbyte
 docker compose up
 ```
 
+```bash
+git clone https://github.com/airbytehq/airbyte.git
+cd airbyte
+env $(cat .env.podman) podman-compose -f docker-compose.podman.yaml up
```
Comment (marcosmarxm) [inline README.md:58]: What happen if you use .env and docker-compose.yaml file instead?

**Review by marcosmarxm — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (jwszolek) [reply README.md:58]: Another issue that arises is the manner in which references to variables are defined.

**Comment by terencecho** [comment]
@marcosmarxm i'll defer to your judgement for this PR. I don't see any dangers in this PR, but an unsure about the tradeoff between more env/docker-compose files vs putting this information into an article/documentation.

**Comment by jwszolek** [comment]
@marcosmarxm  as a developer, I prefer using a podman-compose file that I can run, rather than relying on documentation that requires it to be included. Please let me know about your final vision. All the best!

**Comment by marcosmarxm** [comment]
@davinchia do you mind giving your opinion here?

**Comment by davinchia** [comment]
Thanks for your contribution @jwszolek!

In the long term, our vision is to support as many different types of deployments as possible via tooling as it's relatively easy to generate both docker compose and podman manifests from central configuration files.

 In the short term, resourcing means we can't support as many as we want to. The worst case here being an unsupported alpha deployment being a bad experience for everyone.

Because of this, I'm going to leave this as an open PR for now. Hopefully we'll be able to pick this up in the next quarter or so. Hope you understand!

**Comment by jwszolek** [comment]
Hey @davinchia, no problem at all. We've already forked this branch internally and are maintaining it with podman-compose implementation.

If you're interested, I'd be happy to help identify opportunities to support and continuously improve this podman configuration. This approach might be a great compromise between the long- and short-term options you mentioned earlier.

Looking forward to hearing from you soon!

Thanks,
Jakub

**Comment by davinchia** [comment]
Hi @jwszolek since 1) we've moved platform code to the https://github.com/airbytehq/airbyte-platform repo 2) it's looking unlikely we are going to add Podman support in 2023, I'm going to close this PR instead of migrating it over the the new repo.

Again, thank you for your contribution! We will reach out to you when we add Podman support.

**Comment by vijayradke** [comment]
@jwszolek Are you able to run airbyte using podman? If yes then could you please let me know what changes we need to do for it? I tried by adding your files `docker-compose.podman.yaml` and `.env.podman` in latest code but getting error.

**Comment by marcosmarxm** [comment]
@vijayradke Airbyte removed the support for Docker a few months ago. I recommend to you run Airbyte using abctl or Kubernetes.

**Comment by jwszolek** [comment]
hi @vijayradke Can you share the error message you're getting? I haven't worked with that in a while. Btw. @marcosmarxm is there a specific reason for removing Docker support?

**Comment by marcosmarxm** [comment]
https://github.com/airbytehq/airbyte/discussions/40599

**Comment by vijayradke** [comment]
Hi @jwszolek here is the logs,
**Command ->** env $(cat .env.podman) podman-compose -f docker-compose.podman.yaml up

**Logs:**

41cec49567489ea73f29c4b502490f7bca3a357e490b20ce956582f1a0e53e0e
Resolving "airbyte/init" using unqualified-search registries (/etc/containers/registries.conf.d/999-podman-machine.conf)
Trying to pull docker.io/airbyte/init:0.40.32-podman...
Error: initializing source docker://airbyte/init:0.40.32-podman: reading manifest 0.40.32-podman in docker.io/airbyte/init: manifest unknown
Resolving "airbyte/bootloader" using unqualified-search registries (/etc/containers/registries.conf.d/999-podman-machine.conf)
Trying to pull docker.io/airbyte/bootloader:0.40.32-podman...
Error: initializing source docker://airbyte/bootloader:0.40.32-podman: reading manifest 0.40.32-podman in docker.io/airbyte/bootloader: manifest unknown
Resolving "airbyte/db" using unqualified-search registries (/etc/containers/registries.conf.d/999-podman-machine.conf)
Trying to pull docker.io/airbyte/db:0.40.32-podman...
Error: initializing source docker://airbyte/db:0.40.32-podman: reading manifest 0.40.32-podman in docker.io/airbyte/db: manifest unknown
Resolving "airbyte/temporal" using unqualified-search registries (/etc/containers/registries.conf.d/999-podman-machine.conf)
Trying to pull docker.io/airbyte/temporal:0.40.32-podman...
Error: initializing source docker://airbyte/temporal:0.40.32-podman: reading manifest 0.40.32-podman in docker.io/airbyte/temporal: manifest unknown
Resolving "airbyte/cron" using unqualified-search registries (/etc/containers/registries.conf.d/999-podman-machine.conf)
Trying to pull docker.io/airbyte/cron:0.40.32-podman...
Error: initializing source docker://airbyte/cron:0.40.32-podman: reading manifest 0.40.32-podman in docker.io/airbyte/cron: manifest unknown
Resolving "airbyte/worker" using unqualified-search registries (/etc/containers/registries.conf.d/999-podman-machine.conf)
Trying to pull docker.io/airbyte/worker:0.40.32-podman...
Error: initializing source docker://airbyte/worker:0.40.32-podman: reading manifest 0.40.32-podman in docker.io/airbyte/worker: manifest unknown
Resolving "airbyte/server" using unqualified-search registries (/etc/containers/registries.conf.d/999-podman-machine.conf)
Trying to pull docker.io/airbyte/server:0.40.32-podman...
Error: initializing source docker://airbyte/server:0.40.32-podman: reading manifest 0.40.32-podman in docker.io/airbyte/server: manifest unknown
Resolving "airbyte/connector-builder-server" using unqualified-search registries (/etc/containers/registries.conf.d/999-podman-machine.conf)
Trying to pull docker.io/airbyte/connector-builder-server:0.40.32-podman...
Error: initializing source docker://airbyte/connector-builder-server:0.40.32-podman: reading manifest 0.40.32-podman in docker.io/airbyte/connector-builder-server: manifest unknown
Resolving "airbyte/webapp" using unqualified-search registries (/etc/containers/registries.conf.d/999-podman-machine.conf)
Trying to pull docker.io/airbyte/webapp:0.40.32-podman...
Error: initializing source docker://airbyte/webapp:0.40.32-podman: reading manifest 0.40.32-podman in docker.io/airbyte/webapp: manifest unknown
[init]                             | Error: no container with name or ID "init" found: no such container
[bootloader]                       | Error: no container with name or ID "airbyte-bootloader" found: no such container
[db]                               | Error: no container with name or ID "airbyte-db" found: no such container
[airbyte-temporal]                 | Error: no container with name or ID "airbyte-temporal" found: no such container
[worker]                           | Error: no container with name or ID "airbyte-worker" found: no such container
[airbyte-cron]                     | Error: no container with name or ID "airbyte-cron" found: no such container
[server]                           | Error: no container with name or ID "airbyte-server" found: no such container
[airbyte-connector-builder-server] | Error: no container with name or ID "airbyte-connector-builder-server" found: no such container
[webapp]                           | Error: no container with name or ID "airbyte-webapp" found: no such container

**Comment by vijayradke** [comment]
@marcosmarxm  Thank you for update will try with suggested approach.

**Comment by nehakhanpathan786** [comment]
Hi @jwszolek 
Can we use **Podman** to build up **Airbyte** locally and run it in **windows**? 
Beacuse I have taken clone for 
     git clone https://github.com/airbytehq/airbyte.git
     cd airbyte
      env $(cat .env.podman) podman-compose -f docker-compose.podman.yaml up
But it gives me an error of 

Error: no container with name or ID "airbyte_airbyte-server_1" found: no such container
Error: no container with name or ID "airbyte_airbyte-db_1" found: no such container
Error: no container with ID or name "airbyte_airbyte-db_1" found: no such container
Error: no container with ID or name "airbyte_airbyte-server_1" found: no such container
ba1aff67af76b22536b29c953a31195780080f94d9abcb6ecf83ea3fefa5046e
e7806b59a9f0c2260140aa96a048d66019f1f904130dca2858c284d83dea2159
Resolving "airbyte/airbyte" using unqualified-search registries (/etc/containers/registries.conf.d/999-podman-machine.conf)
Trying to pull docker.io/airbyte/airbyte:latest...
Error: initializing source docker://airbyte/airbyte:latest: reading manifest latest in docker.io/airbyte/airbyte: requested access to the resource is denied
b6ad3eeaa4e8ae59e03174791db129ccc4447749982d8b00a482866de8a7ce8d
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "C:\Users\nehakha\AppData\Local\Programs\Python\Python311\Scripts\podman-compose.exe\__main__.py", line 7, in <module>
  File "C:\Users\nehakha\AppData\Local\Programs\Python\Python311\Lib\site-packages\podman_compose.py", line 3504, in main
    asyncio.run(async_main())
  File "C:\Users\nehakha\AppData\Local\Programs\Python\Python311\Lib\asyncio\runners.py", line 190, in run
    return runner.run(main)
           ^^^^^^^^^^^^^^^^
  File "C:\Users\nehakha\AppData\Local\Programs\Python\Python311\Lib\asyncio\runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\nehakha\AppData\Local\Programs\Python\Python311\Lib\asyncio\base_events.py", line 654, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "C:\Users\nehakha\AppData\Local\Programs\Python\Python311\Lib\site-packages\podman_compose.py", line 3500, in async_main
    await podman_compose.run()
  File "C:\Users\nehakha\AppData\Local\Programs\Python\Python311\Lib\site-packages\podman_compose.py", line 1743, in run
    retcode = await cmd(self, args)
              ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\nehakha\AppData\Local\Programs\Python\Python311\Lib\site-packages\podman_compose.py", line 2521, in compose_up
    loop.add_signal_handler(signal.SIGINT, lambda: [t.cancel("User exit") for t in tasks])
  File "C:\Users\nehakha\AppData\Local\Programs\Python\Python311\Lib\asyncio\events.py", line 578, in add_signal_handler
    raise NotImplementedError
NotImplementedError

beacuse of which I am not able to proceed further and I have also tried this in mac as well as WSL but nothing works it gives me same error

**Comment by nehakhanpathan786** [comment]
Hi @marcosmarxm @jwszolek 
could you please guide me how to Setup and Run Airbute using podman?

**Pull request closed without merging**